text/0000-safe-cancel.md

```diff
@@ -0,0 +1,98 @@
+
+- Title: safe-cancel
+- Authors: [John Tromp](mailto:john.tromp@gmail.com)
+- Start date: Oct 10, 2019
+- RFC PR: Edit if merged: [mimblewimble/grin-rfcs#0000](https://github.com/mimblewimble/grin-rfcs/pull/0000)
+- Tracking issue: [Edit if merged with link to tracking github issue]
+
+---
+
+## Summary
+[summary]: #summary
+
+Allow for safe cancellation of pending transactions, preventing future so-called play attacks.
+
+## Motivation
+[motivation]: #motivation
+
+A wallet cannot simply cancel a pending transaction by forgetting about it and returning its inputs to the wallet balance.
+Especially not when the other party is responsible for completing and broadcasting the transaction.
+They may still do so at any time as long as the inputs are not spent differently.
+
+## Community-level explanation
+[community-level-explanation]: #community-level-explanation
+
+Suppose Bob as the receiver sends Alice as the sender an invoice slatepack
+(Receiver-Sender-Receiver or RSR flow) to which Alice responds, but then Bob
+doesn't finalize it.  He might suggest a problem with the invoice, with his
+wallet, or the exchange rate, or the (suddenly realized)
+need to pay in a different currency. Whatever the case, he convinces Alice to
+have the current transaction cancelled.  Alice is fine with that, but needs to
+make sure that Bob doesn't later complete the transaction and steal Alice's
+inputs.
+
+Alice has two options; spend a transaction input back to herself (a self-spend), or
+construct a new transaction that shares at least one input with the
+old transaction. The former is cleaner, but requires separate fees, and possibly waiting
+for confirmation.
+
+The wallet provides a command for each of these options.
+
+grin-wallet respend [OPTIONS]
+
+will mark the pending transaction, specified either by ID or TxID UUID,
+as requiring one of its inputs to be spent in the very next wallet spend.
+
+grin-wallet unspend [OPTIONS]
```

Comment:
Maybe make `respend/unspend` one sub-command with options to choose the transaction-style + fees. Think this would be clearer to the user, and could even have an interactive UI component explaining the tradeoffs (more of an implementation detail).

Reply:
I'm thinking about this as well. We will have 3 ways to cancel a transaction "cancel, unspend, respend" which might be confusing to the user. Would it make sense to join these commands e.g. `./grin-wallet cancel -unspend` or `./grin-wallet cancel -respend`?

Reply:
and grin-wallet cancel -unsafe to reproduce current behaviour?